<launch>
    <arg name="p_id"/>
    <arg name="object_name"/>
    <arg name="date"/>
    <arg name="data_dir" default="$(find contactdb_utils)/data/contactdb_data"/>
    <arg name="kinect_res" default="qhd"/>
    <arg name="vis" default="true"/>
    <arg name="const_thermal_norm" default="false"/>
    <arg name="play_rate" default="0.75"/>
    <arg name="do_texture_mapping" default="true"/>

    <!-- play bag file -->
    <param name="use_sim_time" value="true"/>
    <node name="player" pkg="contactdb_utils" type="play_bag.sh" output="screen"
          required="true" args="$(arg data_dir) $(arg p_id) $(arg object_name) $(arg play_rate)"/>

    <!-- TF players -->
    <node name="extrinsics_publisher" type="tf_publisher.py"
          pkg="contactdb_utils" output="screen" required="true"
          args="--date $(arg date)"/>
    <node name="object_tf_publisher" pkg="contactdb_utils" output="screen"
          type="optitrack_object_alignment_publisher.py"
          args="--object_name $(arg object_name)" required="true"/>

    <group ns="deepgrasp">
        <!-- uncompress thermal images -->
        <node name="republish_thermal" type="republish" pkg="image_transport"
              output="screen" required="true"
              args="
              compressed in:=boson/image_raw_raw
              raw out:=boson/image_raw_raw"/>

        <!-- AGC and rectify the thermal image -->
        <include file="$(find contactdb_utils)/launch/thermal_proc.launch"
                 ns="boson">
            <arg name="old" value="false"/>
            <arg name="const_norm" value="$(arg const_thermal_norm)"/>
        </include>

        <!-- texture mapping -->
        <node name="texture_mapping" pkg="contactdb_utils"
              type="texture_mapping2.py" output="screen" required="true"
              if="$(arg do_texture_mapping)"
              args="
              --object $(arg object_name)
              --output $(arg data_dir)/$(arg p_id)/$(arg object_name)/$(arg object_name).ply"
              ns="boson">
        </node>
    </group>

    <!-- RViz -->
    <node name="rviz" pkg="rviz" type="rviz" output="screen" required="true"
          args="-d $(find contactdb_utils)/rviz_files/texture_mapping.rviz"
          if="$(arg vis)"/>
</launch>
